controllers/utils/mapper.go

```diff
@@ -23,25 +23,32 @@ func NHCByNodeMapperFunc(c client.Client, logger logr.Logger) handler.MapFunc {
 	// are unrelated to this node. Its even possible that the node still doesn't
 	// have the right labels set to be picked up by the nhc selector.
 	delegate := func(o client.Object) []reconcile.Request {
-		var nhcList remediationv1alpha1.NodeHealthCheckList
-		err := c.List(context.Background(), &nhcList, &client.ListOptions{})
-		if err != nil {
-			return nil
+		requests := make([]reconcile.Request, 0)
+
+		node := &v1.Node{}
+		if err := c.Get(context.Background(), client.ObjectKey{Name: o.GetName()}, node); err != nil {
+			logger.Error(err, "failed to get node", "node name", o.GetName())
+			return requests
+		}
+
+		nhcList := &remediationv1alpha1.NodeHealthCheckList{}
+		if err := c.List(context.Background(), nhcList, &client.ListOptions{}); err != nil {
+			logger.Error(err, "failed to list NHCs")
+			return requests
 		}
-		var r []reconcile.Request
+
 		for _, nhc := range nhcList.Items {
 			selector, err := metav1.LabelSelectorAsSelector(&nhc.Spec.Selector)
 			if err != nil {
-				logger.Error(err, "failed to use the NHC selector.")
+				logger.Error(err, "invalid node selector", "NHC name", nhc.GetName())
+				continue
```

Comment:
IIUC due to the webhook validation this will not occur on CRs which created after the validation is in place, so this fix is relevant for older CRs right ?